app/controllers/notes_controller.rb

```diff
@@ -46,6 +46,27 @@ def index
 		render 'index', formats: [:html]
   end
 
+  def all_notes
+    notes = Note.includes(:user, :noteable).order(created_at: :desc)
+    notes_data = notes.map do |note|
+      {
+        date: note.format_date,
+        user_name: note.user.user_name,
+        display_for: note.noteable.display_for_note,
+        message: note.notes_message,
+        id: note.id,
+        modifiable: allowed_to?(:modify?, note)
+      }
+    end
+
+    column_header = {
```

Comment:
(See above comments, this is just a reminder that you can delete this part. You can then remove one level of nesting from the returned JSON too, which is nice.)